### ISSUE TITLE
Fedora RPM distribution for AFL++

### ISSUE THREAD

**Issue by alexandredoyen29** [AFLplusplus/AFLplusplus#2023]
Hi the AFL++ team !

I write this issue to tell you that I think it is a good idea to develop a .spec file to build a RPM distribution for AFL++ for each releases. It could be also built through the Fedora Copr building system.

On my free time, I could try to develop this file. I think it is not hard (Expect translate DEB requirements to RPM requirements to build AFL++).

Best regards

**Comment by vanhauser-thc** [comment]
If you want to go for it, sure!

**Comment by vanhauser-thc** [comment]
@alexandredoyen29 what is your progress?

**Comment by alexandredoyen29** [comment]
I have a preliminary .spec file that compile AFL++ to a RPM, but I have to test it to find if all features are available.
Because I do this on my free time, it could be long ;)

**Comment by vanhauser-thc** [comment]
Do you think I get paid? ;)

**Comment by kanru** [comment]
A spec file already exist for AFL++ https://src.fedoraproject.org/rpms/american-fuzzy-lop/

Maybe the name is not what you expect?

**Comment by Dedsec0098** [comment]
Hi AFL++ team!

I'm interested in contributing a .spec file to create an RPM package for AFL++. I’ve started translating the DEB requirements to RPM and will test it on Fedora/CentOS to ensure it builds and installs correctly. Once tested, I’ll submit a pull request with instructions on using it for RPM-based distributions and potentially setting it up on Fedora Copr.

Best regards, Shrish Mishra

**Comment by vanhauser-thc** [comment]
I am pretty sure it exists already as package in red hat so a spec file should already exist there

**Comment by alexandredoyen29** [comment]
> I am pretty sure it exists already as package in red hat so a spec file should already exist there

Yep, an american-fuzzy-lop-4.21c-5.fc41.x86_64 RPM already exists. The problem is it is compiled in "source-only" mode, and not in "distrib" mode. So, some modes are unavailable  :/

**Comment by rwmjones** [comment]
You're welcome to help with packaging.  No idea what "source-only" mode means.

**Comment by vanhauser-thc** [comment]
> You're welcome to help with packaging. No idea what "source-only" mode means.

this are `make` targets. `source-only` builds those components that are for source code targets, `binary-only` for fuzzing targets that have no source code, and `distrib` builds all.
`distrib` would be best to have.

**Comment by Jvlegod** [comment]
> Yep, an american-fuzzy-lop-4.21c-5.fc41.x86_64 RPM already exists. The problem is it is compiled in "source-only" mode, and not in "distrib" mode. So, some modes are unavailable :/

i'm curious that how about the progress now? :)

**Comment by rwmjones** [comment]
If you want to change Fedora packaging it's best to discuss this where we do Fedora packaging:
https://src.fedoraproject.org/rpms/american-fuzzy-lop
Ideally by submitting a pull request there to fix whatever you think isn't working.
There is also a mailing list for general Fedora development discussion:
https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/

Not sure if keeping this github bug open serves any useful purpose.
